File: scripts/itc-data.js
#!/usr/bin/env node
/**
 * ITC (IntoTheCryptoverse) Data Fetcher
 * Fetches key macro/risk metrics for weekly market report
 * 
 * Usage: node itc-data.js
 * Output: JSON with risk levels, recession data, macro indicators
 */

const { createSession, navigateTo, extractContent } = require('../skills/web-scout/profiles/itc.js');

async function fetchRiskLevel(page, asset) {
  const navResult = await navigateTo(page, `${asset}-risk`, 4000);
  if (!navResult.ok || navResult.expired) {
    return { asset, ok: false, expired: navResult.expired };
  }
  
  const content = await extractContent(page);
  const text = content.text;
  
  // Parse risk level from page text
  // Look for patterns like "Risk Level: 0.45" or "0.45"
  const riskMatch = text.match(/(?:risk|level)[:\s]*([0-9.]+)/i) 
    || text.match(/([0-9.]{3,5})/);
  
  return {
    asset,
    ok: true,
    url: content.url,
    title: content.title,
    textPreview: text.substring(0, 300),
    riskValue: riskMatch ? parseFloat(riskMatch[1]) : null
  };
}

async function fetchMacroPage(page, pageName, waitMs = 5000) {
  const navResult = await navigateTo(page, pageName, waitMs);
  if (!navResult.ok || navResult.expired) {
    return { page: pageName, ok: false, expired: navResult.expired };
  }
  
  const content = await extractContent(page);
  return {
    page: pageName,
    ok: true,
    url: content.url,
    title: content.title,
    textPreview: content.text.substring(0, 500)
  };
}

async function main() {
  let browser;
  const output = {
    source: 'itc',
    timestamp: new Date().toISOString(),
    ok: true,
    sessionExpired: false,
    data: {}
  };
  
  try {
    const session = await createSession();
    browser = session.browser;
    const page = session.page;
    
    // Check session validity with dashboard first
    const dashCheck = await navigateTo(page, 'dashboard', 3000);
    if (dashCheck.expired) {
      output.ok = false;
      output.sessionExpired = true;
      output.error = 'ITC Firebase session expired - re-login required';
      console.log(JSON.stringify(output, null, 2));
      await browser.close();
      return;
    }
    
    // Fetch BTC risk
    output.data.btcRisk = await fetchRiskLevel(page, 'btc');
    
    // Fetch recession / interest rate data
    output.data.macro = await fetchMacroPage(page, 'interest-rate', 5000);
    
    // Fetch MVRV-Z score
    output.data.mvrvZ = await fetchMacroPage(page, 'mvrv-z', 4000);
    
    // Fetch Fear & Greed (ITC's version)
    output.data.fearGreed = await fetchMacroPage(page, 'fear-greed', 4000);
    
    console.log(JSON.stringify(output, null, 2));
    
  } catch (err) {
    output.ok = false;
    output.error = err.message;
    console.log(JSON.stringify(output, null, 2));
  } finally {
    if (browser) await browser.close();
  }
}

main();

File: scripts/itc-data.py
#!/usr/bin/env python3
"""
ITC Data Fetcher — Python wrapper for weekly market report integration

Usage: python3 itc-data.py
Output: JSON with ITC risk metrics and session status

If session expired, sends Telegram alert and returns graceful error.
"""

import subprocess
import json
import sys
import os

def fetch_itc_data():
    """Fetch ITC data via Node.js script."""
    script_path = os.path.join(os.path.dirname(__file__), 'itc-data.js')
    
    try:
        result = subprocess.run(
            ['node', script_path],
            capture_output=True,
            text=True,
            timeout=120  # 2 min timeout for multiple page loads
        )
        
        if result.returncode != 0:
            return {
                "source": "itc",
                "ok": False,
                "error": f"Script failed: {result.stderr[:200]}"
            }
        
        data = json.loads(result.stdout)
        
        # Check for session expiration
        if data.get('sessionExpired'):
            send_telegram_alert("⚠️ ITC Firebase session expired! Re-login required at https://app.intothecryptoverse.com")
        
        return data
        
    except subprocess.TimeoutExpired:
        return {
            "source": "itc",
            "ok": False,
            "error": "ITC fetch timeout (120s)"
        }
    except json.JSONDecodeError as e:
        return {
            "source": "itc",
            "ok": False,
            "error": f"JSON parse error: {e}"
        }
    except Exception as e:
        return {
            "source": "itc",
            "ok": False,
            "error": str(e)
        }


def send_telegram_alert(message):
    """Send alert via Telegram."""
    try:
        # Use clawdbot sessions_send or direct API
        subprocess.run(
            ['clawdbot', 'send', '--channel', 'telegram', '--to', 'rick', message],
            capture_output=True,
            timeout=10
        )
    except Exception:
        # Fallback: just log it
        print(f"ALERT (Telegram failed): {message}", file=sys.stderr)


if __name__ == '__main__':
    data = fetch_itc_data()
    print(json.dumps(data, indent=2))
